lib/dynamic_type/src/dynamic_type/dynamic_type.h

```diff
@@ -573,18 +573,31 @@ constexpr bool is_dynamic_type_v = is_dynamic_type<T>::value;
     return ret;                                                            \
   }
 
-DEFINE_BINARY_OP(add, +);
-DEFINE_BINARY_OP(minus, -);
-DEFINE_BINARY_OP(mul, *);
-DEFINE_BINARY_OP(div, /);
-DEFINE_BINARY_OP(mod, %);
-DEFINE_BINARY_OP(band, &);
-DEFINE_BINARY_OP(bor, |);
-DEFINE_BINARY_OP(xor, ^);
-DEFINE_BINARY_OP(land, &&);
-DEFINE_BINARY_OP(lor, ||);
-DEFINE_BINARY_OP(lshift, <<);
-DEFINE_BINARY_OP(rshift, >>);
+DEFINE_BINARY_OP(add, +, operator+);
+DEFINE_BINARY_OP(minus, -, operator-);
+DEFINE_BINARY_OP(mul, *, operator*);
+DEFINE_BINARY_OP(div, /, operator/);
+DEFINE_BINARY_OP(mod, %, operator%);
+DEFINE_BINARY_OP(band, &, operator&);
+DEFINE_BINARY_OP(bor, |, operator|);
+DEFINE_BINARY_OP(xor, ^, operator^);
+DEFINE_BINARY_OP(land, &&, operator&&);
+DEFINE_BINARY_OP(lor, ||, operator||);
+DEFINE_BINARY_OP(lshift, <<, operator<<);
+DEFINE_BINARY_OP(rshift, >>, operator>>);
+
+// Not defining comparison operators that returns DynamicType as operator
+// overloading, because we want to leave the operator overloading for comparison
+// operators that returns bool. Instead, we give each operator a function name,
+// so that users can use the function name to call the operator. That is:
+//   dt1 < dt2 --> returns a bool
```

Comment:
(Trust by verify) 

Where is this defined? I only saw `op_name` (thus `lt`) defined above by `DEFINE_BINARY_OP`.

Reply:
It is defined by `DEFINE_COMPARE_OP`, I will update this PR mentioning that.